### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by Itswali**
### IN THIS PR.

- Setup linters
- setup routers
- setup saas
- setup react
- create different components for books

### CODE REVIEW

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @Itswali , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: :red_circle: :red_circle: 

You have submitted a project, that you did not use the [Microverse README template](https://github.com/microverseinc/readme-template), Please make sure that you use the [Microverse Readme](https://github.com/microverseinc/readme-template) for this project and include all required sections.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <>
+      <nav className="navbar">
+        <div className="Title-name">
+          BOOK STORE
+        </div>
+        <div className="nav-links">
+          <Link to="/Books"> Books </Link>
+          <Link to="/Category"> Category </Link>
+        </div>
+      </nav>
```
Comment (AmaduKamara) [inline src/App.js]: [ Optional Changes ]

- [ ] You could have moved this to the Header.js component to serve as your navbar and then import and use that Header component here. That makes your App.js lean and clean. 👍🏽

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+function Book({ title, author }) {
+  return (
+    <span>
+      {title}
+      by
+      {author}
+    </span>
+  );
+}
```
Comment (AmaduKamara) [inline src/components/Book.js]: - [ ] This component is expected to have a button to remove a book. It's expected that a book has a button that when clicked, deletes a book from the list and the button is best to be placed in this component. Kindly add that 👍🏽

File: src/components/BookItem.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      chapter: 'Introduction',
+    },
+  ];
+  return (
+    <ul className="book-container">
+      {books.map((book) => (
+        <li className="Main-content" key={book.id}>
+          <Book title={book.title} author={book.author} />
+          <button type="button">Remove</button>
+        </li>
```
Comment (AmaduKamara) [inline src/components/BookItem.js]: - [ ] Kindly move this Remove button to the Book component. It's not supposed to be here as this component is only rendering a list of books when looped over the array of books and making use of the single book component to render as many books as the array has. Kindly remove this button from here and move it to the Book.js component. 👍🏽
- [ ] Kindly render the CreateBook form component on this page just after the ul tag. It's expected that the form is shown in this book list components where the list of books is shown and then a book form is shown after the list. 👍🏽

File: src/components/Category.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const Cartegory = () => (
+  <div>
+    <h1>Cartegory</h1>
+  </div>
```
Comment (AmaduKamara) [inline src/components/Category.js]: - [ ] For this project, you're to have a button in this component with the text Check Status. This button will do something in the future when clicked. Kindly add a button with the text Check Status  👍🏽

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      <Routes>
+        <Route path="/Category" element={<Cartegory />} />
+        <Route path="/Books" element={<Books />} />
+        <Route path="/" />
```
Comment (AmaduKamara) [inline src/App.js]: - [ ] In the route where you are using the Books component, kindly have the path to just "/" so the homepage route renders the books. Currently, you are having an empty route with "/" that is showing an empty page that does not exist. The goal is to make the "/" route the homepage route and shows all the books. Kindly work on that 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @Itswali,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You sed Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ README is great 👍🏽
✅ Used create-react-app and react-router-dom for project initialization 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Category.js (hunk 4 above)
Comment (Itswali) [reply src/components/Category.js]: Implemented the suggested changes.

File: src/App.js (hunk 5 above)
Comment (Itswali) [reply src/App.js]: Change the path as requested.

File: src/App.js (hunk 1 above)
Comment (Itswali) [reply src/App.js]: Its Done thanks for pointing it out.

File: src/components/Book.js (hunk 2 above)
Comment (Itswali) [reply src/components/Book.js]: Button is added as requested.

File: src/App.js (hunk 5 above)
Comment (Itswali) [reply src/App.js]: Change the links path

File: src/components/Category.js (hunk 4 above)
Comment (Itswali) [reply src/components/Category.js]: Add the button with check status content

File: src/components/BookItem.js (hunk 3 above)
Comment (Itswali) [reply src/components/BookItem.js]: removed the button from book component and added in the create component along with book component as PER REQUESTED.

**Review by pinkmoon25 — APPROVED** [review]
Hi @Itswali , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
linter checks :heavy_check_mark: 
good readme :heavy_check_mark: 
Changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Itswali** [comment]
@PinkMoon25  Thanks for the Review.

**Pull request merged**